src/util/misc.ts

```diff
@@ -347,3 +347,11 @@ export function objToQueryString(obj: KVS<any>): string {
 export function sanitizeString(str: string) {
   return str.replace(/[&<>"'*=//]/g, ' ');
 }
+
+export function getArticleBase(articleId: string) {
+    if (typeof articleId === 'undefined') {
+        return undefined;
+    }
+    // Encoding ? char because it is not allowed directly in hrefs.
+    return articleId.replace('?', encodeURIComponent('?'));
```

Comment:
I still believe this is wrong. I believe this function makes the unencoded article_id ready for a URL, so for example `C♯_(musical_note)` shoudl be probably changed to `C%E2%99%AF_(musical_note)`!?

Reply:
No, it is not required. That's not a "#" character if you see it closely.

First of all, I tried to create the page with # character on wiki, I couldn't. It just wouldn't allow me and trim everything after the hash.

Secondly, if the case appears I have handled it by encoding "#" char as well. But in my opinion, no article id would have "#" in it.